sdk/core/Azure.Core/src/TelemetryDetails.cs

```diff
@@ -161,5 +166,18 @@ private static string EscapeProductInformation(string productInfo)
             sb.Append(')');
             return sb.ToString();
         }
+
+        // Would be nice to use https://learn.microsoft.com/dotnet/api/system.text.ascii.isvalid (.NET 8+) in the future
+        private static bool ContainsNonAscii(string value)
```

Comment:
Curious: did we measure the perf cost of this?  It looks like this is only called at pipeline-construction time, but can you confirm?  If it's true, the perf overhead is not critical, but if we call this each time we do pipeline.Send(message), we might want to look for opportunities to streamline further.

Reply:
I confirmed that this is only called at pipeline construction time. Yeah I agree if this was called on every send this wouldn't work.